_drafts/2021-11-18-journey-to-engineering-blog.md

```diff
@@ -0,0 +1,28 @@
+
+I joined Maxwell in January 2019 as a Sales Development Representative.  I had recently quit my marketing career, completed a web development program and was weighing my options on how to move forward [(have you seen how may there are?)](https://github.com/kamranahmedse/developer-roadmap).  One idea was to try sales engineering, so gaining sales experience at a tech company seemed like a good step.  I was specifically attracted to Maxwell because of the problem they were solving, the why behind it and the company culture of collaboration, hard work and transparency.
+
+From day one I was open and honest with my leadership about my desire to be closer to the technology side of the business and when a slot opened up on the Customer Success Team, I was excited for the opportunity to jump in head first!  During my time in Customer Success I connected with our Director of Engineering who, after finding out I’d completed my program, offered a mentorship I couldn’t pass up.  I started to take on various small projects and bugs, learning our code base and gaining experience along the way.
+
+Today I am a Software Engineer at Maxwell and I couldn’t be more thrilled.  While the path was not linear, there are numerous experiences I gained along the way that I feel set me up for success as an engineer including the benefits of learning the product you’re developing and how communication goes a long way in fostering forward momentum.
+
+### Learn about the product you are developing
+
+Through my time in sales I gathered knowledge of our product, the complexities of the mortgage industry and what our prospects were looking for (or not).  From customer success I learned even more about our platform, how it worked, what customers liked and didn’t like, including how they used the platform in their day to day and just how different those use cases were.  This allowed me to step into the engineering role with a deeper context around the platform from a user perspective which, as a new engineer to the code base, helped immensely in understanding what was going on in the code, the design and product overall.
+
+### Communication goes a long way
+
+Aside from the immediate platform knowledge, I had now been calling prospects and talking to customers and I had gained a ton of confidence in communication I didn’t even know I was lacking.  This felt key to being a successful engineering team member due to the ever changing nature of not just being at a startup, but being at one which is in growth mode!  There is a lot going on, there are a lot of changes happening and it’s important to be able to speak up, reach out and voice opinions.  It may spark a needed conversation, ignite a new idea or you might just learn something!
```

Comment:
Might be worth mentioning something about how prepared you would have been here without your specific experience. 

I can only speak to my experience, but I can imagine without my time as a cashier I'd have been much less willing to reach out with questions in my first dev job. Could be cool to highlight how your experience helped ease the transition into engineering.

Reply:
@regexpressyourself , great suggestion. 

@bethjaw , you might be able to talk about how your experience from CS was a great way to get confidence that details matter in communication. E.g when a customer would report an issue, you learned in CS to ask for as much detail and context as possible. As a software engineer now, you can clearly see how when debugging a customer issue, having the details and context really help get to the bottom of the issue more quickly. 

Another area where your confidence in communication might be helping is when working on migrating existing customers from the legacy proxy setup to the new Cloudflare SSL for SaaS architecture that you have been owning. 

https://developers.cloudflare.com/ssl/ssl-for-saas

Reply:
@regexpressyourself , great suggestion.

@bethjaw , you might be able to talk about how your experience from CS was a great way to get confidence that details matter in communication. E.g when a customer would report an issue, you learned in CS to ask for as much detail and context as possible. As a software engineer now, you can clearly see how when debugging a customer issue, having the details and context really help get to the bottom of the issue more quickly.

Another area where your confidence in communication might be helping is when working on migrating existing customers from the legacy proxy setup to the new Cloudflare SSL for SaaS architecture that you have been owning.

https://developers.cloudflare.com/ssl/ssl-for-saas